src/middlewares/__tests__/createInsightsMiddleware.ts

```diff
@@ -79,22 +104,46 @@ describe('insights', () => {
       });
     });
 
-    it('warns dev if userToken is set before creating the middleware', () => {
-      const { insightsClient, instantSearchInstance } = createTestEnvironment();
-      insightsClient('setUserToken', 'abc');
-      expect(() => {
-        createInsightsMiddleware({
-          insightsClient,
-        })({ instantSearchInstance });
-      })
-        .toWarnDev(`[InstantSearch.js]: You set userToken before \`createInsightsMiddleware()\` and it is ignored.
-Please set the token after the \`createInsightsMiddleware()\` call.
+    it('does not throw when an event is sent right after the creation in UMD', () => {
+      const algoliaAnalytics = createAlgoliaAnalytics();
+      const {
+        insightsClient,
+        libraryLoadedAndProcessQueue,
+        instantSearchInstance,
+      } = createUmdTestEnvironment(algoliaAnalytics);
 
-createInsightsMiddleware({ /* ... */ });
+      const middleware = createInsightsMiddleware({
+        insightsClient,
+      })({ instantSearchInstance });
+      middleware.subscribe();
 
-insightsClient('setUserToken', 'your-user-token');
-// or
-aa('setUserToken', 'your-user-token');`);
+      // It tries to send an event.
+      instantSearchInstance.sendEventToInsights({
```

Comment:
We keep testing internals here, that's fragile. Again, that's how it was before these changes.

Reply:
Although the method is public, in this context, we're indeed testing the internals. I'll keep it that way for now.